f3/granite.go

```diff
@@ -703,7 +622,13 @@ func (q *quorumState) HasAgreement() bool {
 
 // Checks whether at least one message has been received from a strong quorum of senders.
 func (q *quorumState) ReceivedFromStrongQuorum() bool {
-	return q.sendersTotalPower > q.powerTable.Total*2/3
+	two := NewStoragePower(2)
+	three := NewStoragePower(3)
+
+	threshold := new(StoragePower).Mul(q.powerTable.Total, two)
+	threshold.Div(threshold, three)
+
+	return q.sendersTotalPower.Cmp(threshold) > 0
```

Comment:
Can you factor out a `hasStrongQuorum(part, total *StoragePower) bool` pure function to use here and above? And maybe do a `hasWeakQuorum` too for symmetry.

Reply:
done.

Reply:
Remember to replace the code here with a call to your new method

Reply:
Good catch! Done (upcoming commit)